### PR TITLE
CLOUDP-146481: Convert OplogMinRetentionHours field properly

### PR DESCRIPTION
**Pull request by fabritsius**

### CODE REVIEW

File: pkg/api/v1/atlasdeployment_types.go [hunk 1]
```diff
@@ -304,10 +307,28 @@ type ProcessArgs struct {
...
 
+func convertOplogMinRetentionHours(specArgs *ProcessArgs, atlasArgs *mongodbatlas.ProcessArgs) error {
+	if specArgs != nil && specArgs.OplogMinRetentionHours != "" {
+		OplogMinRetentionHours, err := strconv.ParseFloat(specArgs.OplogMinRetentionHours, 64)
+		if err != nil {
+			return err
+		}
+
+		atlasArgs.OplogMinRetentionHours = &OplogMinRetentionHours
+		specArgs.OplogMinRetentionHours = ""
```
Comment (helderjs) [inline pkg/api/v1/atlasdeployment_types.go:326]: Don't you think changing the spec value in that way may lead to side effects? Someone working in this field would not expect it was altered after calling `ToAtlas()`

**Review by helderjs — CHANGES_REQUESTED** [review]

File: pkg/api/v1/atlasdeployment_types.go [hunk 2]
```diff
@@ -328,21 +349,30 @@ func (specArgs ProcessArgs) IsEqual(newArgs interface{}) bool {
...
 			return false
 		}
 	}
 
 	return true
 }
 
+func stringValue(v interface{}) string {
+	return regexp.
+		MustCompile(`\.[0]*$`).
```
Comment (igor-karpukhin) [inline pkg/api/v1/atlasdeployment_types.go:372]: `MustCompile` can be called once, not every time you call this method

**Review by igor-karpukhin — APPROVED** [review]
Looks good, with one minor thing with regex

File: pkg/api/v1/atlasdeployment_types.go [hunk 3]
```diff
@@ -304,10 +307,28 @@ type ProcessArgs struct {
 
 func (specArgs ProcessArgs) ToAtlas() (*mongodbatlas.ProcessArgs, error) {
```
Comment (Sugar-pack) [inline pkg/api/v1/atlasdeployment_types.go:308]: Why do we use value receiver here? Autogenerated methods use pointer receiver. I might be a bit confusing

File: pkg/api/v1/atlasdeployment_types.go (hunk 2 above)
Comment (fabritsius) [reply pkg/api/v1/atlasdeployment_types.go:372]: Good point 👌

File: pkg/api/v1/atlasdeployment_types.go (hunk 1 above)
Comment (fabritsius) [reply pkg/api/v1/atlasdeployment_types.go:326]: Yeah, I agree that is should fine in this case as the changes only affect `ToAtlas()` scope.

**Review by helderjs — APPROVED** [review]
Good job! :+1:

**Review by Sugar-pack — APPROVED** [review]

**Pull request merged**